Description: deuglify mail syntax highlighting
 Undo the mail formatting horror by reverting to the 3.5-2 (lenny)
 version of syntax/mail.jsf.in. Neither the upstream commits nor the
 NEWS file really explain the benefits of the new stuff, I only see
 the bugs and all the excess red that's making me nauseous, too.
Author: Josip Rodin
Bug-Debian: http://bugs.debian.org/597825
Last-Update: 2010-11-13

Index: joe-4.1/syntax/mail.jsf.in
===================================================================
--- joe-4.1.orig/syntax/mail.jsf.in	2015-03-01 14:36:25.000000000 +0100
+++ joe-4.1/syntax/mail.jsf.in	2016-01-31 18:01:34.000000000 +0100
@@ -1,188 +1,52 @@
 # JOE syntax highlight file for typical UNIX mail files
 # 2004-04-25 Tomas Szepe <szepe@pinerecords.com>
 
-# Improved to handle quote characters commonly seen on Usenet, and
-# highlighting of the more frequently edited mail and news headers, with
-# some sanity checking of the same.
-# 2004-06-26 Jeff Hurwit <darkmoon@verizon.net>
-
--
-
 =Idle
-=Head	bold
-=HdTo	red bg_white
-=HdFrom	bold red
-=HdSbj	bold magenta
-=HdDate	bold
-=HdRRT	black bg_white
-=HdGrp	bold white bg_red
-=HdFuT	bold red bg_yellow
-=HdBad	bold red
-=Quot1	green bg_white
-=Quot2	bold blue bg_white
-=Sign	bold magenta
+=Quot1	green
+=Quot2	cyan
+=Sign	magenta
 
-# Start in headers. We know we're out at the first blank line.
-
-:first Idle
-	*		hbol		noeat
+:newline Idle
+	*		knocolor
 	"\n"		newline
-	">|:~}#]"	newline		noeat
-
-# Require header
-#	"\n"		nohead
+	" "		newline
+	"\t"		newline
+	">"		q1trans		noeat
+	"-"		maybesign1
 
-:nohead HdBad
-	*		nohead
-
-:hbol Idle
-	*		hbad		recolor=-1
+:maybesign1 Idle
+	*		knocolor
 	"\n"		newline
-	"A-Z"		bufhdrs		buffer
-	"-"		newline		noeat
+	"-"		maybesign2
 
-:hbad HdBad
-	*		hbad
-	"\n"		hbol
-
-:hlbad HdBad
-	*		hbol
-
-:bufhdrs Idle
-	*		hbad		recolor=-1
-	"\n"		hlbad		noeat recolor=-2
-	"-_a-zA-Z0-9"	bufhdrs
-	":"		headers		hold
-
-:headers Idle
-	*		hbad		recolor=-1
-	" \n"		hnocolor	noeat strings
-	"To"		hsto		recolor=-2
-	"Cc"		hsto		recolor=-2
-	"Bcc"		hsto		recolor=-2
-	"From"		hsfrom		recolor=-2
-	"Subject"	hssbj		recolor=-2
-	"Date"		hsdate		recolor=-2
-	"Return-Receipt-To"	hsrrecp	recolor=-2
-	"Newsgroups"	hsgroups	recolor=-2
-	"Followup-To"	hsfolup		recolor=-2
-done
-
-:hnocolor Idle
-	*		hnocolor
-	"\n"		cont_nocolor
-
-:cont_nocolor Idle
-	*		hbol		noeat
-	" \t"		hnocolor	recolor=-1
-
-:hsto Head
-	*		hto
-	"\n"		hbol
-
-:hto HdTo
-	*		hto
-	"\n"		cont_to
-
-:cont_to HdTo
-	*		hbol		noeat
-	" \t"		hto		recolor=-1
-
-:hsfrom Head
-	*		hfrom
-	"\n"		hbol
-
-:hfrom HdFrom
-	*		hfrom
-	"\n"		hbol
-
-:hssbj Head
-	*		hsbj
-	"\n"		hbol
-
-:hsbj HdSbj
-	*		hsbj
-	"\n"		cont_sbj
-
-:cont_sbj HdSbj
-	*		hbol		noeat
-	" \t"		hsbj		recolor=-1
-
-:hsdate Head
-	*		hdate
-	"\n"		hbol
-
-:hdate HdDate
-	*		hdate
-	"\n"		hbol
-
-:hsrrecp Head
-	*		hrrecp
-	"\n"		hbol
-
-:hrrecp HdRRT
-	*		hrrecp
-	"\n"		hbol
-
-:hsgroups Head
-	*		hgroups
-	"\n"		hbol
-
-:hgroups HdGrp
-	*		hgroups
-	"\n"		hbol
-
-:hsfolup Head
-	*		hfolup
-	"\n"		hbol
-
-:hfolup HdFuT
-	*		hfolup
-	"\n"		hbol
-
-# body of the message
-
-:newline Idle
+:maybesign2 Idle
 	*		knocolor
 	"\n"		newline
-	">|:~}#]"	q1		recolor=-1
-	"-"		maybesign1	recolor=-1
-
-:maybesign1 Quot1
-	*		q1
-	" \t"		q1
-	"\n"		newline		recolor=-1
-	">|:~}#]"	q2		recolor=-1
-	"-"		maybesign2	recolor=-1
+	" "		maybesign3
 
-:maybesign2 Quot2
-	*		q2
-	"\t"		q2
-	"\n"		newline		recolor=-1
-	" "		maybesign3	recolor=-1
-	">|:~}#]"	q1		recolor=-1
-	"-"		knocolor	recolor=-3
-
-:maybesign3 Quot2
-	*		q2
-	" \t"		q2
-	"\n"		sign		recolor=-4
-	"->|:~}#]"	q1		recolor=-1
+:maybesign3 Idle
+	*		knocolor
+	"\n"		sign
 
 :sign Sign
 	*		sign
 
+:q1trans Quot1
+	*		q1		recolor=-1
+
 :q1 Quot1
 	*		kq1
 	"\n"		newline
-	" \t"		q1
-	"->|:~}#]"	q2		recolor=-1
+	" "		q1
+	"\t"		q1
+	">"		q2		recolor=-1
 
 :q2 Quot2
 	*		kq2
 	"\n"		newline
-	" \t"		q2
-	"->|:~}#]"	q1		recolor=-1
+	" "		q2
+	"\t"		q2
+	">"		q1		recolor=-1
 
 :knocolor Idle
 	*		knocolor
